src/terminal.rs

```diff
@@ -32,12 +81,48 @@ pub struct TerminalOptions {
     pub viewport: Viewport,
 }
 
-/// Interface to the terminal backed by a [`Backend`].
+/// An interface that Ratatui to interact and draw [`Frame`]s on the user's terminal.
+///
+/// This is the main entry point for Ratatui. It is responsible for drawing and maintaining the
+/// state of the buffers, cursor and viewport.
+///
+/// The [`Terminal`] is generic over a [`Backend`] implementation which is used to interface with
+/// the underlying terminal library. The [`Backend`] trait is implemented for three popular Rust
+/// terminal libraries: [Crossterm], [Termion] and [Termwiz]. See the [`backend`] module for more
+/// information.
+///
+/// The terminal has two buffers that are used to draw the application. The first buffer is the
+/// current buffer and the second buffer is the previous buffer. The two buffers are compared at
+/// the end of each draw pass to output the necessary updates to the terminal.
```

Comment:
```suggestion
/// the end of each draw pass to output the only the changes to the terminal.
```

This seems like good context to mention the on a terminal resize event, the developer using ratatui should clear and render everything. Also that there's a `Clear` widget that lets a developer clear an `Rect` area.

Reply:
The draw() method calls `autoresize()` to handle this, but the app should call draw. I added this to the doc.